test/es-module/test-esm-resolve-type.js

```diff
@@ -98,86 +100,148 @@ try {
-   *  returning the information as specified above. Source for 'url' value
-   *  is [1], source for 'format' value is [2]
-   */
+  function testDualPackageWithJsMainScriptAndModuleType() {
+    // Create a dummy dual package
+    //
+    /**
+     * this creates the following directory structure:
+     *
+     * ./node_modules:
+     *   |-> my-dual-package
+     *       |-> es
+     *           |-> index.js
+     *           |-> package.json [2]
+     *       |-> lib
+     *           |-> index.js
+     *       |->package.json [1]
+     *
+     * in case the package is imported:
+     *    import * as my-package from 'my-dual-package'
+     * it will cause the resolve method to return:
+     *  {
+     *     url: '<base_path>/node_modules/my-dual-package/es/index.js',
+     *     format: 'module'
+     *  }
+     *
+     *  following testcase ensures that resolve works correctly in this case
+     *  returning the information as specified above. Source for 'url' value
+     *  is [1], source for 'format' value is [2]
+     */
+
+    const moduleName = 'my-dual-package';
 
-  const moduleName = 'my-dual-package';
-
-  const mDir = rel(`node_modules/${moduleName}`);
-  const esSubDir = rel(`node_modules/${moduleName}/es`);
-  const cjsSubDir = rel(`node_modules/${moduleName}/lib`);
-  const pkg = rel(`node_modules/${moduleName}/package.json`);
-  const esmPkg = rel(`node_modules/${moduleName}/es/package.json`);
-  const esScript = rel(`node_modules/${moduleName}/es/index.js`);
-  const cjsScript = rel(`node_modules/${moduleName}/lib/index.js`);
-
-  createDir(nmDir);
-  createDir(mDir);
-  createDir(esSubDir);
-  createDir(cjsSubDir);
-
-  const mainPkgJsonContent = {
-    type: 'commonjs',
-    main: 'lib/index.js',
-    exports: {
-      '.': {
-        'require': './lib/index.js',
-        'import': './es/index.js'
-      },
-      './package.json': './package.json',
-    }
-  };
-  const esmPkgJsonContent = {
-    type: 'module'
-  };
+    const mDir = rel(`node_modules/${moduleName}`);
+    const esSubDir = rel(`node_modules/${moduleName}/es`);
+    const cjsSubDir = rel(`node_modules/${moduleName}/lib`);
+    const pkg = rel(`node_modules/${moduleName}/package.json`);
+    const esmPkg = rel(`node_modules/${moduleName}/es/package.json`);
+    const esScript = rel(`node_modules/${moduleName}/es/index.js`);
+    const cjsScript = rel(`node_modules/${moduleName}/lib/index.js`);
 
-  fs.writeFileSync(pkg, JSON.stringify(mainPkgJsonContent));
-  fs.writeFileSync(esmPkg, JSON.stringify(esmPkgJsonContent));
-  fs.writeFileSync(esScript,
-                   'export function esm-resolve-tester() {return 42}');
-  fs.writeFileSync(cjsScript,
-                   `module.exports = { 
-                     esm-resolve-tester: () => {return 42}}`
-  );
+    createDir(nmDir);
+    createDir(mDir);
+    createDir(esSubDir);
+    createDir(cjsSubDir);
+
+    const mainPkgJsonContent = {
+      type: 'commonjs',
+      main: 'lib/index.js',
```

Comment:
the proposal would be to renounce at the `main` all together? I did that as it does not impact the test but this example was built originally to simulate the situation found in `highlight.js`'s `package.json` that being the main reason why this is here.

Reply:
If we remove `"main"`, let’s also remove `"default"` and `"require"`, which are not used by this test either. Or let’s keep all of them so the test is closer to real world implementations 🤷‍♂️

Reply:
Ah, true. My comment was just a nit, so no change needed. If it were me, I would KISS and remove the unused bits, but maybe not worth the effort.

Reply:
Personally, I would go with test scenarios as close to real world scenarios as possible (would mean having `main` defined in `package.json` even if it is currently not used internally because it is used in _real life_ and would be good to make sure those will not break in the future either). But I really have no strong opinion on this. Just let me know how you think this should be done.